### DISCUSSION TITLE
How to turn off custom callbacks for timeupdate

### DISCUSSION THREAD

**Discussion by kobe651jp** [videojs/video.js#7984] [Q&A] [answered]
I'm trying to add callback function to timeupdate,after that I need take off the callback.

My code as follows
```
                  player.on('timeupdate',()=>{
                    console.log('currentTime',player.currentTime());
                    return true;
                  })
```
try 1:not working
```
                  player.on('timeupdate',()=>{
                    return;
                  })
```
try2: I can turn off the callback event, but the playtime won't update during playback.
`      player.off('timeupdate');
`
Is there a way to safely turn off custom callbacks?

**Comment by gkatsev** [comment, marked as answer]
You need to keep a reference to your method to be able to remove the handler. `player.off('timeupdate')` will remove *all* `timeupdate` handlers.
```js
const timeupdateHandler = () => {
  console.log('currentTime',player.currentTime());
};

player.on('timeupdate', timeupdateHandler);
player.off('timeupdate', timeupdateHandler);
```
Hope this helps.

**Reply by kobe651jp** [reply]
It's works.Thank you!

**Reply by gkatsev** [reply]
you're welcome.
